catroid/src/main/java/org/catrobat/catroid/ui/SettingsActivity.java

```diff
@@ -384,14 +384,16 @@ public static boolean isRaspiSharedPreferenceEnabled(Context context) {
 		return getBooleanSharedPreference(false, SETTINGS_SHOW_RASPI_BRICKS, context);
 	}
 
-	public static void setAutoCrashReportingEnabled(Context context, boolean isEnabled) {
+	public static boolean setAutoCrashReportingEnabled(Context context, boolean isEnabled) {
```

Comment:
Adding a return value just for testing should always be prevented. Additionally setters with a return value are always problematic. You never know what they are returning.

Stateful static classes like the CrashReporter are always hard to test. If code uses such classes, it even gets more complicated.

I've to do more research on this topic (maybe it's better to do DI and use interfaces? Don't know yet :laughing:) but my first thought to test it is. If you come up with something better, that would be great!
* Disable crash reporting in the setup process
* Logging an exception fails
* Enable crash reporting
* Successfully log an exception